hoodie-common/src/main/java/com/uber/hoodie/common/util/SerializationUtils.java

```diff
@@ -16,156 +16,147 @@
+      kryo.setRegistrationRequired(false);
+    }
+
+    byte[] serialize(Object obj) throws IOException {
+      kryo.reset();
+      baos.reset();
+      Output output = new Output(baos);
+      this.kryo.writeClassAndObject(output, obj);
+      output.close();
+      return baos.toByteArray();
+    }
+
+    Object deserialize(byte[] objectData) {
+      return this.kryo.readClassAndObject(new Input(objectData));
     }
   }
 
   /**
-   * <p>
-   * Deserializes a single {@code Object} from an array of bytes.
-   * </p>
-   *
-   * <p>
-   * If the call site incorrectly types the return value, a {@link ClassCastException} is thrown from the call site.
-   * Without Generics in this declaration, the call site must type cast and can cause the same ClassCastException.
-   * Note that in both cases, the ClassCastException is in the call site, not in this method.
-   * </p>
-   *
-   * @param <T> the object type to be deserialized
-   * @param objectData the serialized object, must not be null
-   * @return the deserialized object
-   * @throws IllegalArgumentException if {@code objectData} is {@code null}
-   * @throws HoodieSerializationException (runtime) if the serialization fails
+   * This class has a no-arg constructor, suitable for use with reflection instantiation.
+   * For Details checkout com.twitter.chill.KryoBase.
    */
-  public static <T> T deserialize(final byte[] objectData) {
-    if (objectData == null) {
-      throw new IllegalArgumentException("The byte[] must not be null");
+  private static class KryoInstantiator implements Serializable {
+
+    public Kryo newKryo() {
+
+      Kryo kryo = new KryoBase();
+      // ensure that kryo doesn't fail if classes are not registered with kryo.
+      kryo.setRegistrationRequired(false);
+      kryo.setInstantiatorStrategy(new org.objenesis.strategy.StdInstantiatorStrategy());
+      // Handle cases where we may have an odd classloader setup like with libjars
+      // for hadoop
+      kryo.setClassLoader(Thread.currentThread().getContextClassLoader());
+      return kryo;
+    }
+
+    private static class KryoBase extends Kryo {
+      @Override
+      protected Serializer newDefaultSerializer(Class type) {
+        final Serializer serializer = super.newDefaultSerializer(type);
+        if (serializer instanceof FieldSerializer) {
+          final FieldSerializer fieldSerializer = (FieldSerializer) serializer;
+          fieldSerializer.setIgnoreSyntheticFields(true);
+        }
+        return serializer;
+      }
+
+      @Override
+      protected ObjectInstantiator newInstantiator(Class type) {
+        return () -> {
+          // First try reflectasm - it is fastest way to instantiate an object.
+          try {
+            final ConstructorAccess access = ConstructorAccess.get(type);
+            return access.newInstance();
+          } catch (Throwable t) {
+            // ignore this exception. We may want to try other way.
```

Comment:
Can you add warn logs here and below in the catch ?

Reply:
I am worried it would be too many. Not adding therefore.